### PR TITLE
KNN-CAD algorithm

### PR DESCRIPTION
**Pull request by ish-vlad**
In this branch we provide k-Nearest Neighbour Conformal Anomaly Detection (KNN-CAD) method.
The mathematical explanation of algorithm is available in this [article](https://arxiv.org/abs/1608.04585) (if you will use the algorithm, please make a reference).
Also, you can find brief description [here](https://github.com/ish-vlad/NAB/tree/master/nab/detectors/knncad). 
If you have any questions, please feel free to ask here or send them to ishvlad28@gmail.com.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,3 +1,5 @@
+if you want to use KNN-CAD algorithm, please make reference to our [article](https://arxiv.org/abs/1608.04585).
```
Comment (BoltzmannBrain) [inline README.md]: Please remove this line. I will make a note to this effect in the README after merging this PR.

File: results/final_results.json [hunk 2]
```diff
@@ -1,8 +1,13 @@
 {
     "bayesChangePt": {
-        "reward_low_FN_rate": 32.256563780459771,
+        "reward_low_FN_rate": 32.25656378045977,
```
Comment (BoltzmannBrain) [inline results/final_results.json]: Please remove this change, it's not relevant to this PR.

File: nab/detectors/knn/windowedGaussian_detector.py [hunk 3]
```diff
@@ -0,0 +1,79 @@
+# ----------------------------------------------------------------------
```
Comment (BoltzmannBrain) [inline nab/detectors/knn/windowedGaussian_detector.py]: I'm unsure why this file is included. Please remove it from this PR.

**Comment by BoltzmannBrain** [comment]
@ish-vlad please address my comments in this PR, as well as the merge conflicts, and then we can merge it into NAB 😄 . Sorry for the delay on this.

File: nab/detectors/knncad/knncad_detector.py [hunk 4]
```diff
@@ -0,0 +1,75 @@
+#################################################
```
Comment (BoltzmannBrain) [inline nab/detectors/knncad/knncad_detector.py]: FYI when this is merged I will add [the Numenta license comments](https://github.com/numenta/NAB/blob/master/nab/detectors/numenta/numenta_detector.py#L1-L19) above.

**Comment by ish-vlad** [comment]
@BoltzmannBrain Complete :) 
Сould you please paste the link according to your requirements? Thank you in advance.

**Comment by oxtopus** [comment]
Closing in favor of #281 in which merge conflicts have been resolved.

**Pull request closed without merging**